CHANGELOG.md

```diff
@@ -6,9 +6,142 @@
+allowing to iterate on code from previous cells or even sessions
+(depending on how specific kernel stores history).
+
+To cycle between history items press <kbd>Alt</kbd> + <kbd>Arrow Up</kbd> and  <kbd>Alt</kbd> + <kbd>Arrow Down</kbd>.
+
+To enable execution history, go to Settings → Notebook → check the "Kernel history access" checkbox.
+
+This feature was already available in the console in previous releases; it only works with kernels supporting execution history requests.
+To clear the execution history consult the documentation of the kernel you are using (e.g. IPython/ipykernel).
+
+### Opening Files from Tracebacks
+
+Paths to code files detected in the tracebacks returned by kernels on execution error are now turned into links.
+These links will open the corresponding file for edition (if these are in the jupyter root directory)
+or open a read-only preview (if the file is outside of the root directory and the active kernel supports the debugger).
+
+<img alt="A link to the file with definition of imported module is shown in the traceback" src="https://raw.githubusercontent.com/krassowski/jupyterlab/changelog-4.1/docs/source/getting_started/changelog_assets/4.1-traceback-links.png" class="jp-screenshot">
+
+
+### Error Indicator in the Table of Contents
+
+When cell fails during execution an error indicator will be displayed by the corresponding heading,
+increasing awareness of the notebook state and enabling users to quickly navigate to the cell which requires attention.
+
+<img alt="Error indicator in the shape of a triangle with exclamation sign (using ⚠ UTF character) shown next to one of the heading in the table of contents panel" src="https://raw.githubusercontent.com/krassowski/jupyterlab/changelog-4.1/docs/source/getting_started/changelog_assets/4.1-toc-error-indicator.png" class="jp-screenshot">
+
+### Plugin Manager
+
+Individual plugins can now be disabled or enabled from a new Plugin Manager user interface.
+Each extension is composed of one or more plugins, and plugins form the basis of JupyterLab itself,
+thus the plugin manager enables more extensive customization of the JupyterLab experience.
+
+<img alt="An example search result in the plugin extension listing" src="https://jupyterlab.readthedocs.io/en/latest/_images/plugin-manager-search-notebook.png" class="jp-screenshot">
+
+This feature is intended for advanced users and documented in depth in the [documentation](https://jupyterlab.readthedocs.io/en/latest/user/extensions.html#managing-plugins-with-plugin-manager).
+
+Administrators may want to [lock specific plugins](https://jupyterlab.readthedocs.io/en/latest/user/extensions.html#locking-and-unlocking-plugins) if they are required for any reason; this will prevent users from disabling the plugins via Plugin Manager and remote API calls. The Plugin Manager itself can be [disabled using CLI](https://jupyterlab.readthedocs.io/en/latest/user/extensions.html#enabling-and-disabling-extensions).
```

Comment:
```suggestion
Administrators may want to [lock specific plugins](https://jupyterlab.readthedocs.io/en/latest/user/extensions.html#locking-and-unlocking-plugins) if they are required for any reason; this will prevent users from disabling the plugins via Plugin Manager and remote API calls. The Plugin Manager itself can be [disabled using the CLI](https://jupyterlab.readthedocs.io/en/latest/user/extensions.html#enabling-and-disabling-extensions).
```